### ISSUE TITLE
Throw an exception when an instance detects its own expired locks

### ISSUE THREAD

**Issue by dalaro** [thinkaurelius/titan#892]
Expired locks are logged at `WARN` without respect to the graph instance ID that originally held the now-expired lock.

The code that warns about and drops these expired lock columns should check whether the instance ID matches the reader's own instance ID.  If the expired lock belongs to the running instance, this should propagate up to the user as an exception with a message specifically about lock expiration.  The current implementation throws an exception, but the message is "No lock columns found for KeyColumn".  This message is problematic because it's unnecessarily general -- it has multiple causes, of which a lock expiring while held is one -- and because it's utterly opaque and unactionable from a user standpoint.

Reading an expired claim with an instance ID != self instance ID should still just warn and not throw an exception, the way it works now.

**Comment by dalaro** [comment]
I changed ConsistentKeyLockerTest to expect an exception specifically of type `ExpiredLockException`, a subclass of `TemporaryLockingException`, when a held lock expires in 6ea5ebf8d0b13c025d08a4734037fb725d0e089c.

I changed ConsistentKeyLocker to match the new test expectations in 24f93bbdfff7b20a0e11bfd747813ba1a48111cb.

I'm going to leave this alone for day or two to let it fall out of my short term memory, then come back and review the diffs before closing.
